acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
-				return h.DockerRmi(dockerCli, runImage, buildImage, value)
-			})
+			appName = fmt.Sprintf("some-org/%s", h.RandString(10))
+			appImageName = registry.RepoName(appName)
+
+			assertAppImage = assert.NewImageAssertionManager(appImageName, containerManager)
+
+			mixedComponentMessenger = components.NewMixedComponents(t, testLifecycle, pack)
+
+			cacheCleanupRequired = true
+		})
+
+		it.After(func() {
+			if cacheCleanupRequired {
+				h.DockerRmi(dockerCli, appImageName)
+				ref, err := name.ParseReference(appImageName, name.WeakValidation)
+				assert.Nil(err)
+				cacheImage := cache.NewImageCache(ref, dockerCli)
+				buildCacheVolume := cache.NewVolumeCache(ref, "build", dockerCli)
+				launchCacheVolume := cache.NewVolumeCache(ref, "launch", dockerCli)
+				cacheImage.Clear(context.TODO())
+				buildCacheVolume.Clear(context.TODO())
+				launchCacheVolume.Clear(context.TODO())
+			}
+		})
+
+		it("creates a runnable, rebuildable image on daemon from app dir", func() {
+			stackManager.EnsureDefaultStackCreated()
+			testBuilder := builderManager.EnsureComboBuilderExists()
+			pack.SuccessfulRun("set-default-builder", testBuilder.Name())
+			if pack.Supports("trust-builder") {
+				pack.SuccessfulRun("trust-builder", testBuilder.Name())
+			}
+
+			appPath := filepath.Join("testdata", "mock_app")
+			output := pack.SuccessfulRunWithOutput("build", appImageName, "-p", appPath)
+			cleanupAppImageTask := containerManager.CleanupTaskForImageByName(appImageName)
+			defer cleanupAppImageTask()
+
+			assertOutput := assert.NewOutputAssertionManager(output)
+
+			assertOutput.ReportsSuccessfulImageBuild(appImageName)
+			assertOutput.ReportsUsingBuildCacheVolume()
+			assertOutput.ReportsSelectingRunImageMirror(stackManager.RunImageMirror())
+
+			assertAppImage.RunsMockAppWithOutput("Launch Dep Contents", "Cached Dep Contents")
+			assertAppImage.HasBase(managers.RunImage)
+			assertAppImage.HasRunImageMetadata(managers.RunImage, stackManager.RunImageMirror())
+
+			assert.NoImageExistsInRegistry(appName, registry)
+
+			t.Log("add a local mirror")
+			localRunImageMirror := registry.RepoName("pack-test/run-mirror")
+			assert.Nil(dockerCli.ImageTag(context.Background(), managers.RunImage, localRunImageMirror))
+			defer containerManager.RemoveImages(localRunImageMirror)
+			pack.SuccessfulRun("set-run-image-mirrors", managers.RunImage, "-m", localRunImageMirror)
+
+			t.Log("rebuild")
+			output = pack.SuccessfulRunWithOutput("build", appImageName, "-p", appPath)
+			cleanupRebuiltAppImageTask := containerManager.CleanupTaskForImageByName(appImageName)
+			defer cleanupRebuiltAppImageTask()
+
+			assertOutput = assert.NewOutputAssertionManager(output)
+			assertOutput.ReportsSuccessfulImageBuild(appImageName)
+			assertOutput.ReportsSelectingRunImageMirrorFromLocalConfig(localRunImageMirror)
+
+			assertMixedOutput := assert.NewMixedComponentOutputAssertionManager(output, mixedComponentMessenger)
```

Comment:
The "mixed output" part is a bit hard to follow.

Reply:
Yes! It's a complicated concept, basically the expectations are based on both the `pack` and `lifecycle` versions we're using. I'd love to work out how to make that clearer. Any thoughts on a better name? Or perhaps the limited number of times this is used make it a candidate for being explicit and repeating this complexity in our test cases?